scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java

```diff
@@ -1282,4 +1282,16 @@ public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
         this.jobInfo.setScheduledTimeForRemoval(scheduledTimeForRemoval);
     }
 
+    @Override
+    public void setVariables(Map<String, JobVariable> variables) {
+        super.setVariables(variables);
+        this.jobInfo.setVariables(this.getVariablesAsReplacementMap());
+    }
+
+    @Override
+    public void setGenericInformation(Map<String, String> genericInformation) {
+        super.setGenericInformation(genericInformation);
+        this.jobInfo.setGenericInformation(genericInformation);
+    }
+
```

Comment:
Now, both variables and generic information are propagated to all tasks and available through getTaskInfo().getJobInfo().getVariables() (or getGenericInformation()).